File: claude-usage-widget/android/app/src/main/kotlin/com/sidbh/claudeusage/widget/WidgetState.kt
package com.sidbh.claudeusage.widget

import java.time.Instant

sealed class WidgetState {
    data object Unconfigured : WidgetState()
    data object Loading : WidgetState()
    data object AuthError : WidgetState()
    data class NetworkError(val message: String) : WidgetState()
    data class Loaded(
        val fiveHourPct: Double,
        val fiveHourResetsAt: Instant?,
        val sevenDayPct: Double?, // null = no 7-day data at all
        val sevenDayResetsAt: Instant?,
        val sevenDayLabel: String, // "7-day Opus" when opus is tracked, else "7-day"
        val fetchedAt: Instant,
        val mock: Boolean,
    ) : WidgetState()
}

File: claude-usage-widget-ios/ClaudeUsage/Shared/WidgetState.swift
import Foundation

/// What the widget renders. The store persists `Loaded` values across launches via App Group
/// UserDefaults so the widget extension can read the last good state instantly during reload.
enum WidgetState: Equatable {
    case unconfigured
    case loading
    case authError
    case networkError(String)
    case loaded(Loaded)

    struct Loaded: Equatable, Codable {
        var fiveHourPct: Double
        var fiveHourResetsAt: Date?
        /// `nil` means the 7-day bucket wasn't present at all; render an em-dash.
        var sevenDayPct: Double?
        var sevenDayResetsAt: Date?
        /// "7-day Opus" when the Opus-specific bucket exists; "7-day" when falling back.
        var sevenDayLabel: String
        var fetchedAt: Date
        var mock: Bool
    }
}
